ocaml/libs/stunnel/stunnel.ml

```diff
@@ -257,7 +257,7 @@ let disconnect_with_pid ?(wait = true) ?(force = false) pid =
           ( if wait then
               Forkhelpers.waitpid
           else
-            Forkhelpers.waitpid_nohang
+            Forkhelpers.waitpid_nohang ~close:true
```

Comment:
This is I think the risky part (it should fix the other bug about an FD leak here, or it might introduce an EBADF bug here).

Perhaps split this into 2 commits? One that just does everything except this line (and we can test that and merge it as low risk), and another commit that adds this line which will need a closer review and more wider testing?